### PR TITLE
testing/synapse: new aport

### PR DESCRIPTION
**Pull request by maxice8**
https://github.com/matrix-org/synapse
Matrix reference homeserver

### CODE REVIEW

**Comment by maxice8** [comment]
needs changes to #9829

**Comment by Cogitri** [comment]
Would be nice if it also provided an initd script like https://www.br0tkasten.de/knowledge/matrix-chat-server

**Comment by Cogitri** [comment]
And a synapse user

File: testing/synapse/synapse.initd [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+command_args="-m synapse.app.homeserver --config=$config $command_args"
+
+depends() {
+	need net
+}
+
+start_pre() {
+	# /var/lib/synapse needs to be owned by the synapse user and so does the config location
+	checkpath --directory --owner synapse:synapse /var/lib/synapse
+	checkpath --directory --owner synapse:synapse $config
```
Comment (Cogitri) [inline testing/synapse/synapse.initd]: --file

File: testing/synapse/synapse.initd [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/sbin/openrc-run
+
+name="synapse"
+command="/usr/bin/python3"
+command_args="-m synapse.app.homeserver --config=$config $command_args"
```
Comment (Cogitri) [inline testing/synapse/synapse.initd]: --config-file

File: testing/synapse/synapse.initd (hunk 1 above)
Comment (Cogitri) [reply testing/synapse/synapse.initd]: Should synapse really own the config and not root?

Comment (maxice8) [reply testing/synapse/synapse.initd]: yes

File: testing/synapse/synapse.post-install [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+if [ ! -e etc/synapse/homeserver.yaml ]; then
+	python3 -m synapse.app.homeserver \
+		--server-name my.domain.name \
+		--config-path /etc/synapse/homeserver.yaml \
+		--generate-config \
+		--report-stats=yes
```
Comment (Cogitri) [inline testing/synapse/synapse.post-install]: Ah, shouldn't this do `2>/dev/null` too?

**Pull request closed without merging**